flink-container/src/main/java/org/apache/flink/container/entrypoint/StandaloneJobClusterConfigurationParserFactory.java

```diff
@@ -82,7 +86,7 @@ public StandaloneJobClusterConfiguration createResult(@Nonnull CommandLine comma
 		final int restPort = getRestPort(commandLine);
 		final String hostname = commandLine.getOptionValue(HOST_OPTION.getOpt());
 		final SavepointRestoreSettings savepointRestoreSettings = CliFrontendParser.createSavepointRestoreSettings(commandLine);
-		final JobID jobId = getJobId(commandLine);
+		final JobID jobId = getJobId(commandLine, getGlobalConfiguration(configDir, dynamicProperties));
```

Comment:
I think we should move the decision which `JobID` to use out of the `StandaloneJobClusterConfigurationParserFactory`. That way we don't have to load the global configuration here which actually should only happen after the `StandaloneJobClusterConfiguration` has been parsed giving the configuration directory. The decision could be moved to `StandaloneJobClusterEntryPoint` by introducing a `createJobIdForCluster(clusterConfiguration.getOptionalJobId(), globalConfiguration)`

Reply:
Done